pac4j-oauth/src/main/java/org/pac4j/oauth/client/GenericOAuth20Client.java

```diff
@@ -0,0 +1,149 @@
+package org.pac4j.oauth.client;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.utils.OAuthEncoder;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.oauth.profile.generic.GenericOAuth20Profile;
+import org.pac4j.oauth.profile.generic.GenericOAuth20ProfileDefinition;
+
+/**
+ * <p>This class is a generic OAuth2 client to authenticate users in a standard OAuth2 server.</p>
+ * <p>All configuration parameters can be specified setting the corresponding attribute.</p>
+ * <p>It returns a {@link org.pac4j.oauth.profile.generic.GenericOAuth20Profile}.</p>
+ *
+ * @author Julio Arrebola
+ */
+public class GenericOAuth20Client extends OAuth20Client<GenericOAuth20Profile> {
```

Comment:
Yes, is the OAuth20Client+profile+definition may be used as a generic client, all changes in this pr should be moved to this class.